heat/core/statistics.py

```diff
@@ -1227,7 +1230,7 @@ def reduce_vars_elementwise(output_shape_i):
 
         var_shape = list(var.shape) if list(var.shape) else [1]
 
-        var_tot = factories.zeros(([x.comm.size, 2] + var_shape), device=x.device)
+        var_tot = factories.zeros(([x.comm.size, 2] + var_shape), dtype=x.dtype, device=x.device)
         n_tot = factories.zeros(x.comm.size, device=x.device)
```

Comment:
possible dtype problem here.

Reply:
is var() supposed to return a float32? Even if x is float64?

Reply:
it should not. i guess i missed a couple dtype calls here. can you add them for me? it should be just in this spot

Reply:
That's what I thought, so is this the possible dtype problem you were talking about? I guess I just misunderstood your first comment

Reply:
i just also realized that the dtype of n_tot doesnt matter because it is only used internally